### ISSUE TITLE
Do not read `config.json` for the `complete` commands. Breaks builds

### ISSUE THREAD

**Issue by imsofi** [gorilla-devs/ferium#139]
Currently the `ferium complete $shell` depends on both `config.json` existing and having at least read and write permissions on it. Even if it does not use any data from the file to run.

This is problematic as during packaging ferium you do not have access to `/home/$user/`. Leading to ferium returning `Permission denied (os error 13)` for the missing permissions.

Currently one workaround would be shipping a minimal `config.json` file with `+rw` permissions to allow `complete` to run. But this should not be intended behaviour as it is quite a hack.

## To Reproduce
1. Change to a system account
2. Attempt to run `ferium complete bash`
3. See permission denied error as system accounts do not have `/home/$user` directories.

## Software version
- OS (macOS/Windows/Linux) Linux --> Fedora 35
- Ferium Version (output of `ferium -V`)  ferium 4.1.4

**Comment by imsofi** [comment]
Related to #42

**Comment by KyleUltimate** [comment]
https://github.com/gorilla-devs/ferium/issues/131
Duplicate and already fixed
Update ferium

**Comment by imsofi** [comment]
Updated to 4.1.4, `ferium complete bash` still requires the existence of a read/write-able `config.json` file. But the requirement for a profile to exist in said file is gone.

Main issue is still present that for packaging, ferium requires a minimal `config.json` file to be shipped with read/write permissions to let the `ferium complete` command run.

**Comment by theRookieCoder** [comment]
This problem is actually caused by #87

**Comment by imsofi** [comment]
> This problem is actually caused by #87, in fact I'd consider this a duplicate of that

That issue is about issues first creating the file.

This issue is that you will get a `Permission denied (os error 13)` as you are not allowed to create a `config.json` at all. A well as that the dependency on the `config.json` file for the `complete` command is odd as it does not require any any data from it.

When running `ferium --help` a config file is not required to exist or be created to run without error. I propose to make `ferium complete` also not have this requirement as it's often run under build environments to get competition files populated.

Sorry for not making that clear in the original issue.

**Comment by theRookieCoder** [comment]
> This issue is that you will get a `Permission denied (os error 13)` as you are not allowed to create a `config.json` at all

Huh that's weird, why is that so.

**Comment by imsofi** [comment]
Ferium attempts to both read and write to `/home/$user/.config/ferium/config.json`. Under most build environments you do not use a local user to build the software, therefore a home folder would not exist for that account.

**Comment by imsofi** [comment]
This is still an issue on 4.1.6.

Libiums still is getting loaded in for the subcommand `complete`. Trying to both read and write to a `/homeless-shelter/.config/ferium/config.json`.

I think this may be the relevant part of the code where it gets problematic due to the expectation of a writable home environment.

https://github.com/gorilla-devs/ferium/blob/ae29dbc2528bf944e4c5dc0e3bb7afda4a06ab51/src/main.rs#L77-L83

<details>
<summary>part of a strace output from running `ferium complete bash` in a build environment</summary>

```
ferium> rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
ferium> mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7ffff7acc000
ferium> mprotect(0x7ffff7acd000, 2097152, PROT_READ|PROT_WRITE) = 0
ferium> rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
ferium> clone3({flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, child_tid=0x7ffff7ccc910, parent_tid=0x7ffff7ccc910, exit_signal=0, stack=0x7ffff7acc000, stack_size=0x1ffd40, tls=0x7ffff7ccc640} => {parent_tid=[6715]}, 88) = 6715
ferium> rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7ffff78c8000
ferium> mprotect(0x7ffff78c9000, 2097152, PROT_READ|PROT_WRITE) = 0
ferium> rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
ferium> clone3({flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, child_tid=0x7ffff7ac8910, parent_tid=0x7ffff7ac8910, exit_signal=0, stack=0x7ffff78c8000, stack_size=0x1ffd40, tls=0x7ffff7ac8640} => {parent_tid=[6716]}, 88) = 6716
ferium> rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7ffff76c4000
ferium> mprotect(0x7ffff76c5000, 2097152, PROT_READ|PROT_WRITE) = 0
ferium> rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
ferium> clone3({flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, child_tid=0x7ffff78c4910, parent_tid=0x7ffff78c4910, exit_signal=0, stack=0x7ffff76c4000, stack_size=0x1ffd40, tls=0x7ffff78c4640} => {parent_tid=[6717]}, 88) = 6717
ferium> rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7ffff74c0000
ferium> mprotect(0x7ffff74c1000, 2097152, PROT_READ|PROT_WRITE) = 0
ferium> rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
ferium> clone3({flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, child_tid=0x7ffff76c0910, parent_tid=0x7ffff76c0910, exit_signal=0, stack=0x7ffff74c0000, stack_size=0x1ffd40, tls=0x7ffff76c0640} => {parent_tid=[6718]}, 88) = 6718
ferium> rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> brk(0x555556ece000)                     = 0x555556ece000
ferium> brk(0x555556ef1000)                     = 0x555556ef1000
ferium> statx(0, NULL, AT_STATX_SYNC_AS_STAT, STATX_ALL, NULL) = -1 EFAULT (Bad address)
ferium> statx(AT_FDCWD, "/homeless-shelter/.config/ferium/config.json", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffffffd5780) = -1 ENOENT (No such file or directory)
ferium> mmap(NULL, 2101248, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_STACK, -1, 0) = 0x7ffff72bc000
ferium> mprotect(0x7ffff72bd000, 2097152, PROT_READ|PROT_WRITE) = 0
ferium> rt_sigprocmask(SIG_BLOCK, ~[], [], 8)   = 0
ferium> clone3({flags=CLONE_VM|CLONE_FS|CLONE_FILES|CLONE_SIGHAND|CLONE_THREAD|CLONE_SYSVSEM|CLONE_SETTLS|CLONE_PARENT_SETTID|CLONE_CHILD_CLEARTID, child_tid=0x7ffff74bc910, parent_tid=0x7ffff74bc910, exit_signal=0, stack=0x7ffff72bc000, stack_size=0x1ffd40, tls=0x7ffff74bc640} => {parent_tid=[6719]}, 88) = 6719
ferium> rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> futex(0x555556e9e768, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, NULL, FUTEX_BITSET_MATCH_ANY) = 0
ferium> futex(0x555556e9dc00, FUTEX_WAKE_PRIVATE, 1) = 0
ferium> ioctl(1, TCGETS, 0x7fffffffa420)        = -1 ENOTTY (Inappropriate ioctl for device)
ferium> write(2, "Permission denied (os error 13)", 31Permission denied (os error 13)) = 31
ferium> write(2, "\n", 1
ferium> )                       = 1
ferium> write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
ferium> futex(0x555556e8a1b8, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> futex(0x555556e8a2e8, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> futex(0x555556e8a418, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> futex(0x555556ea0bc8, FUTEX_WAKE_PRIVATE, 2147483647) = 5
ferium> futex(0x555556e89c70, FUTEX_WAKE_PRIVATE, 1) = 1
ferium> close(3)                                = 0
ferium> close(5)                                = 0
ferium> close(4)                                = 0
ferium> sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=8192}, NULL) = 0
ferium> munmap(0x7ffff7ccd000, 12288)           = 0
ferium> exit_group(1)                           = ?
ferium> +++ exited with 1 +++
```
</details>

**Comment by theRookieCoder** [comment]
Ah crap rookie mistake, I forgot to return from the main function after finishing the complete subcommand's code

**Comment by imsofi** [comment]
For a current workaround, i am using `ferium --config-file=config.json complete ...` together with a shell `config.json` file spessifically made to have the permissions `0644`:

```json
{
  "active_profile": 0,
  "active_modpack": 0,
  "profiles": [],
  "modpacks": []
}
```

This is an undesirable hack one has to do, but it does work and lets one run the command in a build environment.

**Comment by imsofi** [comment]
> Ah crap rookie mistake, I forgot to return from the main function after finishing the complete subcommand's code

Haha its always the simplest mistakes that get you. xD

**Comment by theRookieCoder** [comment]
I'll close this since you've created a related issue

**Comment by imsofi** [comment]
I would still keep this open, as the issue still exists. If i do not have a writable home folder. `v4.1.6` will actually error with a `permission denied` error, its not until i give a writable config.json file it will give the `unreachable code` error. So these are two issues.

**Comment by theRookieCoder** [comment]
Oh yes I see the issue you described is something else

**Comment by theRookieCoder** [comment]
Yet another rookie mistake, I forgot to move the config file code as well
